src/libraries/sendtohelixhelp.proj

```diff
@@ -233,6 +232,24 @@
     </ItemGroup>
   </Target>
 
+  <PropertyGroup>
+    <IsTargetOSSupportedByXUnitLogChecker Condition="'$(TargetOS)' != 'browser' and '$(TargetOS)' != 'wasi' and '$(TargetOS)' != 'ios' and '$(TargetOS)' != 'iossimulator' and '$(TargetOS)' != 'tvos' and '$(TargetOS)' != 'tvossimulator' and '$(TargetOS)' != 'maccatalyst' and '$(TargetOS)' != 'android'">true</IsTargetOSSupportedByXUnitLogChecker>
+  </PropertyGroup>
+
+  <Target Name="BuildXUnitLogChecker" BeforeTargets="BuildHelixWorkItems" Condition="'$(IsTargetOSSupportedByXUnitLogChecker)' == 'true'">
+
+    <PropertyGroup>
+      <XUnitLogCheckerDirectory>$([MSBuild]::NormalizeDirectory('$(ArtifactsDir)tests/Common/XUnitLogChecker'))</XUnitLogCheckerDirectory>
+    </PropertyGroup>
+    
+    <MSBuild Projects="$(MSBuildThisFileDirectory)../tests/Common/XUnitLogChecker/XUnitLogChecker.csproj" Targets="Restore;Build" Properties="Configuration=Release;OutDir=$(XUnitLogCheckerDirectory)" />
```

Comment:
There are couple of issues:
1. The Restore and Build target must run with different evaluations. This means that you should pass in a different global property for restore: https://github.com/dotnet/runtime/blob/655b177da2e423a04a60a331bc6c11fe5ec427c3/src/libraries/testPackages/testPackages.proj#L153C42-L153C42
2. This doesn't work with static graph restore / build. We require our projects to do a upfront restore so that in case of a failure, we fail immediately, up-front. Avoid this MSBuild call alltogether by adding a ProjectReference to this project instead and let MSBuild do its thing. This might be better placed in sendtohelix.proj as sendtohelixhelp runs per inner invocation.

Reply:
I still need to address this but I'll ask you more details about it tomorrow.